spec/services/subscriptions/dates/monthly_service_spec.rb

```diff
@@ -247,6 +252,76 @@
           expect(result).to eq(subscription.started_at.to_date.to_s)
         end
       end
+
+      context 'when plan is pay in advance' do
+        let(:pay_in_advance) { true }
+        let(:subscription_date) { DateTime.parse('02 Feb 2020') }
+
+        it 'returns the start of the previous period' do
+          expect(result).to eq('2022-02-02')
```

Comment:
Is it correct? It's not the previous period in that case, but the current, no?

Reply:
@rsempe, thanks for the review!

It looks correct:
We are in the test of `charge_from_date` method. Charges are billed on the finished previous period no matter if the plan is payed in advance or in arrears. 
The current period case would apply for the `from_date` method